### PR TITLE
Compact some code

### PR DESCRIPTION
**Pull request by eumiro**
1. access dict.keys() implicitly
2. `range(0, x)` can be written as `range(x)`, simplify some loops and iterate directly over the elements

### CODE REVIEW

File: fpdf/fpdf.py [hunk 1]
```diff
@@ -451,8 +450,7 @@ def get_string_width(self, s, normalized=False):
                 else:
                     w += 500
         else:
-            for i in range(0, l):
-                w += cw.get(s[i], 0)
+            w += sum(cw.get(char, 0) for char in s)
```
Comment (Lucas-C) [inline fpdf/fpdf.py:453]: 👍 Nice one

File: tutorial/tuto5.py [hunk 2]
```diff
@@ -28,8 +28,8 @@ def improved_table(self, header, data):
         # Column widths
         w = [40, 35, 40, 45]
         # Header
-        for i in range(0, len(header)):
-            self.cell(w[i], 7, header[i], 1, 0, "C")
+        for w_, header_ in zip(w, header):
+            self.cell(w_, 7, header_, 1, 0, "C")
```
Comment (Lucas-C) [inline tutorial/tuto5.py:32]: `width` and `header_text` or `header_str` would be more meaningful variable names here

File: tutorial/tuto5.py [hunk 3]
```diff
@@ -51,8 +51,8 @@ def fancy_table(self, header, data):
         self.set_font(style="B")
         # Header
         w = [40, 35, 40, 45]
-        for i in range(0, len(header)):
-            self.cell(w[i], 7, header[i], 1, 0, "C", 1)
+        for w_, header_ in zip(w, header):
+            self.cell(w_, 7, header_, 1, 0, "C", 1)
```
Comment (Lucas-C) [inline tutorial/tuto5.py:55]: Same remark for the variable names

File: fpdf/fpdf.py [hunk 4]
```diff
@@ -2332,15 +2328,15 @@ def interleaved2of5(self, txt, x, y, w=1, h=10):
...
                 raise RuntimeError('Char "%s" invalid for I25: ' % char_bar)
-            if not char_space in bar_char.keys():
+            if char_space not in bar_char:
                 raise RuntimeError('Char "%s" invalid for I25: ' % char_space)
 
             # create a wide/narrow-seq (first digit=bars, second digit=spaces)
-            seq = ""
-            for s in range(0, len(bar_char[char_bar])):
-                seq += bar_char[char_bar][s] + bar_char[char_space][s]
+            seq = "".join(
```
Comment (Lucas-C) [inline fpdf/fpdf.py:2337]: 👍 Much cleaner

File: fpdf/fpdf.py [hunk 5]
```diff
@@ -679,7 +677,7 @@ def add_font(self, family, style="", fname=None, uni=False):
                     if self.diffs[i] == diff:
                         d = i
                         break
-                if d == 0:
+                else:
```
Comment (Lucas-C) [inline fpdf/fpdf.py:680]: Could you maybe use this opportunity to replace the `for`-loop by a call to `any()`?

**Comment by Lucas-C** [comment]
You can also use the opportunity of this PR to add yourself to the `CONTRIBUTORS.md` file if you wish :)

File: fpdf/fpdf.py (hunk 5 above)
Comment (eumiro) [reply fpdf/fpdf.py:680]: I am not sure how to convert this to `any()`. If it searches for the first matching element (and needs its index), then it should be a `next()`. But that would span over several lines, so this way it is probably easier to follow.

I'll leave it this way (fix the other things though) and please feel free to change it later.

**Comment by eumiro** [comment]
Okay, @Lucas-C I am done with this PR.

**Review by Lucas-C — APPROVED** [review]

**Comment by Lucas-C** [comment]
Thank you for your contribution!

**Comment by alexanderankin** [comment]
Looks great

File: fpdf/fpdf.py (hunk 5 above)
Comment (eumiro) [reply fpdf/fpdf.py:664]: @Lucas-C Sorry, I am still returning to this one:

Does this `else` ever get executed?

https://github.com/PyFPDF/fpdf2/blob/9fe0f46c1c11855c07ddccfb022fbd45353053d1/fpdf/fpdf.py#L632-L638

The `for` loop goes from `1` to `len(self.diff) + 1`, so the last iteration has `i = len(self.diff)` and therefore there should be an `IndexError` in the line 633. So if this code runs, it has to `break` way before the loop is over.

What was your idea with the `any()`?

Comment (Lucas-C) [reply fpdf/fpdf.py:664]: Sadly this code is currently not covered by any test: https://codecov.io/gh/PyFPDF/fpdf2/src/master/fpdf/fpdf.py

I opened this issue to add a unit test covering this part of code: https://github.com/PyFPDF/fpdf2/issues/92

At the time of my previous comment, I just thought that by using `any()` we could avoid a 4-lines for-loop, without digging much more in the code.

Now, I'm sure this part of the code could be improved for readibility, but I'm more concerned about having non-regression tests covering this usage of `add_font`

**Pull request merged**